_source/_posts/2021-10-04-spring-boot-spa.md

```diff
@@ -0,0 +1,649 @@
+
+```bash
+npm install --save \
+@okta/okta-signin-widget \
+@okta/okta-vue \
+@okta/okta-auth-js \
+axios
+```
+
+Create a `src/okta/index.js` file:
+
+```js
+import OktaSignIn from '@okta/okta-signin-widget'
+import { OktaAuth } from '@okta/okta-auth-js'
+
+const yourOktaUri = '${yourOktaUri}';
+const clientId = '${clientId}';
+
+const oktaSignIn = new OktaSignIn({
+    baseUrl: yourOktaUri,
+    clientId: clientId,
+    redirectUri: 'http://localhost:8080/login/callback',
+    authParams: {
+        pkce: true,
+        issuer: `${yourOktaUri}/oauth2/default`,
+        display: 'page',
+        scopes: ['openid', 'profile', 'email']
+    }
+});
+
+const oktaAuth = new OktaAuth({
+    issuer: `${yourOktaUri}/oauth2/default`,
+    clientId: clientId,
+    redirectUri: window.location.origin + '/login/callback',
+    scopes: ['openid', 'profile', 'email']
+})
+
+export { oktaAuth, oktaSignIn };
+```
+
+In the file above, **you need to replace the placeholders `${...}` with the appropriate values**. Before you can do this, however, you need to use the CLI to create the OIDC application on the Okta servers, which you will do directly below.
+
+## Create A Single-Page OIDC Application On Okta
+
+This time you're going to create a single-page application for the front-end (instead of the web application type you used for the resource server). Enter the following command into a shell.
+
+```bash
+okta apps create
+```
+
+Press `enter` to accept the default **Application name** `okta-app`. (Application name here refers to the OIDC application created on the Okta servers).
+
+For **Type of Application**, Select `2: Single Page App`.
+
+Change the **Redirect URI** to `http://localhost:8080/login/callback`.
+
+Press `enter` to accept the default **Post Logout Redirect**.
+
+You should see some output like the following.
+
+```bash
+Okta application configuration: 
+Issuer:    https://{yourOktaUri}/oauth2/default
+Client ID: {clientId}
+```
+
+The values in brackets will, of course, be your actual values.
+
+**You need to go back to the `src/okta/index.js` file and replace the `yourOktaUri` and `clientId` variables at the top of the file with your values.** The value for `yourOktaUri` is the Okta URI without any further path specifiers. It will look like this: `https://dev-133337.okta.com`
+
+One of the things that the Okta CLI does for you is to add your application's base URL to the CORS trusted origins for the Okta auth server. This is necessary because the the Okta Sign-In Widget will be making cross0-origin requests, that, as noted above, would be blocked unless CORS is properly handled. You can see this by going to your Okta developer dashboard and selected **Security** and **API** from the left-side menu and selected **Trusted Origins** from the API tab. You'll see that (in this case) `http://localhost:8080` was added as a trusted origin of type `CORS Redirect`.
```

Comment:
```suggestion
One of the things that the Okta CLI does for you is to add your application's base URL to the CORS trusted origins for the Okta auth server. This is necessary because the the Okta Sign-In Widget will be making cross-origin requests, that, as noted above, would be blocked unless CORS is properly handled. You can see this by going to your Okta developer dashboard. with **Security** and **API** selected from the left-side menu, and **Trusted Origins** selected from the API tab. You'll see that (in this case) `http://localhost:8080` was added as a trusted origin of type `CORS Redirect`.
```

Reply:
Adding back in @havi's suggestion